### PR TITLE
ORIGIN opcode

### PR DESCRIPTION
**Pull request by ChihChengLiang**
Spec https://github.com/appliedzkp/zkevm-specs/pull/128

_Edit(1)_
- [x] Update Origin opcode tests to use `TestContext::simple_ctx_with_bytecode` once https://github.com/appliedzkp/zkevm-circuits/pull/422 is merged.

### CODE REVIEW

**Comment by ChihChengLiang** [comment]
This is blocked by mocking the tx sender address. Will hold until #349 is completed.

**Review by ed255 — APPROVED** [review]
LGTM!  Modulo the inclusion of the test update from https://github.com/appliedzkp/zkevm-circuits/pull/422

File: zkevm-circuits/src/evm_circuit/execution/origin.rs
Comment (han0110) [inline zkevm-circuits/src/evm_circuit/execution/origin.rs:108]: The market is not used currently and could be removed. (But won't affect anything so it's totally fine)

**Review by han0110 — APPROVED** [review]
LTGM!

File: zkevm-circuits/src/evm_circuit/execution/origin.rs
Comment (CPerezz) [reply zkevm-circuits/src/evm_circuit/execution/origin.rs:108]: Thanks for the catch!!!!

**Pull request merged**
